src/position-manager/AllowancePositionManager.sol

```diff
@@ -0,0 +1,222 @@
+import {PositionManagerBase} from 'src/position-manager/PositionManagerBase.sol';
+import {IAllowancePositionManager} from 'src/position-manager/interfaces/IAllowancePositionManager.sol';
+
+/// @title AllowancePositionManager
+/// @author Aave Labs
+/// @notice Position manager to handle withdraw permit, credit delegation and borrow actions on behalf of users.
+contract AllowancePositionManager is
+  IAllowancePositionManager,
+  PositionManagerBase,
+  NoncesKeyed,
+  EIP712
+{
+  using SafeERC20 for IERC20;
+  using EIP712Hash for *;
+  using MathUtils for uint256;
+
+  /// @notice Mapping of withdraw allowances.
+  mapping(address owner => mapping(address spender => mapping(uint256 reserveId => uint256 amount)))
+    private _withdrawAllowances;
+
+  /// @notice Mapping of credit delegations.
+  mapping(address owner => mapping(address spender => mapping(uint256 reserveId => uint256 amount)))
+    private _creditDelegations;
+
+  /// @dev Constructor.
+  /// @param spoke_ The address of the spoke contract.
+  constructor(address spoke_) PositionManagerBase(spoke_) {}
+
+  /// @inheritdoc IAllowancePositionManager
+  function approveWithdraw(address spender, uint256 reserveId, uint256 amount) external {
+    _updateWithdrawAllowance(msg.sender, spender, reserveId, amount, true);
+  }
+
+  /// @inheritdoc IAllowancePositionManager
+  function approveWithdrawWithSig(
+    EIP712Types.WithdrawPermit calldata params,
+    bytes calldata signature
+  ) external {
+    require(block.timestamp <= params.deadline, InvalidSignature());
+    bytes32 digest = _hashTypedData(params.hash());
+    require(
+      SignatureChecker.isValidSignatureNow(params.owner, digest, signature),
+      InvalidSignature()
+    );
+    _useCheckedNonce(params.owner, params.nonce);
+
+    _updateWithdrawAllowance(params.owner, params.spender, params.reserveId, params.amount, true);
+  }
+
+  /// @inheritdoc IAllowancePositionManager
+  function creditDelegation(address spender, uint256 reserveId, uint256 amount) external {
+    _updateCreditDelegation(msg.sender, spender, reserveId, amount, true);
+  }
+
+  /// @inheritdoc IAllowancePositionManager
+  function creditDelegationWithSig(
+    EIP712Types.CreditDelegation calldata params,
+    bytes calldata signature
+  ) external {
+    require(block.timestamp <= params.deadline, InvalidSignature());
+    bytes32 digest = _hashTypedData(params.hash());
+    require(
+      SignatureChecker.isValidSignatureNow(params.owner, digest, signature),
+      InvalidSignature()
+    );
+    _useCheckedNonce(params.owner, params.nonce);
+
+    _updateCreditDelegation(params.owner, params.spender, params.reserveId, params.amount, true);
+  }
+
+  /// @inheritdoc IAllowancePositionManager
+  function renounceWithdrawAllowance(address owner, uint256 reserveId) external {
+    _updateWithdrawAllowance(
+      owner,
+      msg.sender,
+      reserveId,
+      0,
+      !(_withdrawAllowances[owner][msg.sender][reserveId] == 0)
```

Comment:
emit event arg should not be passed but rather evaluated inside the internal method i think whenever there is a state change 

also, the ask for emitting event only on change was mainly for renounce methods & not really on methods controlled by user - from stermi suggestion - such that a ui which tracks notifs can't be phished by artistry caller renouncing themselves as posm